src/megameklab/com/ui/BattleArmor/tabs/EquipmentTab.java

```diff
@@ -789,20 +789,30 @@ public int compare(String s0, String s1) {
                 return -1;
             } else {
                 //get the numbers associated with each string
-                float r1 = parseDamage(s1);
-                float r0 = parseDamage(s0);
-                return ((Comparable<Float>)r1).compareTo(r0);
+                double r1 = parseDamage(s1);
+                double r0 = parseDamage(s0);
+                return ((Comparable<Double>)r1).compareTo(r0);
             }
         }
 
-        private float parseDamage(String s) {
-            float damage = 0;
-            if(s.contains("/")) {
-                damage = Float.parseFloat(s.split("/")[0]);
+        /**
+         * Extracts a numeric value from the damage string for use in sorting by damage. If a String
+         * contains any non-digit character, that character and anything after it is ignored. If the string
+         * starts with a non-digit character the return value is 0.
+         * 
+         * @param s A weapon damage string
+         * @return  The value to use for sorting.
+         */
+        private double parseDamage(String s) {
+            s = s.replaceAll("[^0-9\\.]+", "/");
+            if (s.contains("/")) {
+                s = s.substring(0, s.indexOf("/"));
+            }
+            if (s.length() > 0) {
+                return Double.parseDouble(s);
```

Comment:
This would be more likely to choke than the `Integer.parseInt` above, would it be better to `try...catch` with a warning log and return 0? Would keep the UI trucking along and the log would still be preserved.

Reply:
There's no need to log anything. There are perfectly valid input strings that do not contain any numeric value, such as "Special". What this does is check for those and return zero for sorting purposes. Any other value is truncated at the first non-digit character before parsing.